### PR TITLE
Feature/cs/focus on feedback

### PR DESCRIPTION
**Pull request by cjshawMIT**
When using a screenreader:

* Should read the question counter (X of Y) on click of the "Next / Previous" buttons.
* Should shift focus to the overall question `div` on click of the "Next / Previous" buttons.
* Should shift focus to the feedback area after clicking "Check Answer" button.

### CODE REVIEW

File: client/js/_player/components/assessments/item.jsx [hunk 1]
```diff
@@ -3,6 +3,18 @@ import _                       from 'lodash';
...
 import UniversalInput          from './universal_input';
 
+
+// TODO:
+//   1) Need to handle the case for no response selected,
+//      and make sure it doesn't re-read "Please select an answer"
+//      when navigating between questions.
+//      * IDEA: Set a prop in assessment.jsx, for "navigated"?
+//              Can we already do this? So if navigating to this Item, do not
+//              set the focus on the feedback element?
```
Comment (resource11) [inline client/js/_player/components/assessments/item.jsx]: Is focus to the feedback item happening only after a user submits an answer?

File: client/js/_player/components/assessments/item.jsx [hunk 2]
```diff
@@ -40,6 +52,8 @@ export default class Item extends React.Component {
     audioPause        : React.PropTypes.func.isRequired,
     audioRecordStart  : React.PropTypes.func.isRequired,
     audioRecordStop   : React.PropTypes.func.isRequired,
+
+    sendSize          : React.PropTypes.func
```
Comment (resource11) [inline client/js/_player/components/assessments/item.jsx]: what is this prop for?

File: client/js/_player/components/assessments/item.jsx (hunk 1 above)
Comment (cjshawMIT) [reply client/js/_player/components/assessments/item.jsx]: Ah, I should have updated my list of TODO's. No, the focus to the feedback also happens the first time they try to submit no answer (just click the "Check Answer" button). I don't think there is an easy way to switch the focus on multiple "no answer" clicks, without adding convoluted logic somewhere...

File: client/js/_player/components/assessments/item.jsx (hunk 2 above)
Comment (cjshawMIT) [reply client/js/_player/components/assessments/item.jsx]: I don't know, it was an existing prop. I added the newline before it when I was experimenting with stuff, but not the prop itself.

Comment (cjshawMIT) [reply client/js/_player/components/assessments/item.jsx]: I'm happy to remove the newline and put it back the way it was....

Comment (cjshawMIT) [reply client/js/_player/components/assessments/item.jsx]: Oh, I remember now. It's an existing prop, but linting complains that it's not in props validation. So I just stuck it in the list.

File: client/js/_player/components/assessments/item.jsx [hunk 3]
```diff
@@ -94,20 +108,39 @@ export default class Item extends React.Component {
...
+  componentWillReceiveProps(nextProps) {
+    // Check if the component has gone from an answer-checking
+    //   state to a done-answer-checking state.
+    // This indicates we should set
+    //   focus on the ``feedback`` ref, so that it can be
+    //   read aloud by screen readers.
+    // The first set of conditionals work for submitting responses.
+    // The second set of conditionals work for the first no-response submission,
+    //   because ``numQuestionsChecking`` does not toggle in that case.
+    // TODO: How to handle multiple / repeated no-response submissions?
```
Comment (resource11) [inline client/js/_player/components/assessments/item.jsx]: We're handling multiple/repeated no-response submissions by adding landmarks to the assessment container/sub containers.

File: client/js/_player/components/assessments/item.jsx [hunk 4]
```diff
@@ -3,6 +3,17 @@ import _                       from 'lodash';
...
+
+// TODO:
+//   1) Think about how to handle the case for when the student
+//      submits "no answer" multiple times (i.e. A -> B -> A).
+//      Should we shift focus to the feedback element each time?
+//      (Currently we handle just the first case).
+//      It's not clear that handling this is actually possible, but is
+//      something we should think about. Because when someone clicks
+//      "Check Answer" multiple times, the component actually doesn't
+//      change, so it's not re-rendered...
```
Comment (resource11) [inline client/js/_player/components/assessments/item.jsx]: We should just handle the first case. We're going to be adding some landmarks into the assessment so the user can more easily jump between the header, question section, answer section, answer feedback, and assessment nav regions.

**Pull request merged**